src/api/scrape.ts

```diff
@@ -0,0 +1,140 @@
+        const ogImageMatch = response.data.match(/<meta[^>]*property="og:image"[^>]*content="([^"]*)"/)
+        if (ogImageMatch && ogImageMatch[1]) {
+            const ogImageUrl = resolveUrl(ogImageMatch[1])
+            console.log(`Found OG image for ${company}:`, ogImageUrl)
+
+            // Verify the OG image URL is accessible
+            try {
+                await axios.head(ogImageUrl)
+                return ogImageUrl
+            } catch (e) {
+                console.log(`OG image not accessible for ${company}, trying next option`)
+            }
+        }
+
+        // Try direct favicon.ico as last resort
+        const directFaviconUrl = `${baseUrl}/favicon.ico`
+        try {
+            await axios.head(directFaviconUrl)
+            console.log(`Found direct favicon.ico for ${company}:`, directFaviconUrl)
+            return directFaviconUrl
+        } catch (error) {
+            console.log(`No direct favicon.ico found for ${company}`)
+        }
+
+        console.log(`No logo found for ${company}`)
+        return null
+    } catch (error) {
+        console.error(`Error fetching logo for ${company}:`, error)
+        return null
+    }
+}
+
+/**
+ * @param {import('next').NextApiRequest} req
+ * @param {import('next').NextApiResponse} res
+ */
+export default async function scrape(req, res) {
+    if (req.method !== 'GET') {
+        res.setHeader('Allow', 'GET')
+        res.status(405).json({ success: false, message: 'Method not allowed' })
+        return
+    }
+
+    const company = req.query.company || 'supabase'
+
+    try {
+        // Get jobs data
+        const response = await axios.get(`https://jobs.ashbyhq.com/${company}`)
+        const match = response.data.match(/window\.__appData\s*=\s*({[^;]+})/)
+        if (!match) {
+            throw new Error(`Could not find job data in response for ${company}`)
+        }
+
+        const jobData = JSON.parse(match[1])
+        const jobs = jobData?.jobBoard?.jobPostings || []
+
+        // Try to get company logo
+        const logo = await getCompanyLogo(company)
+
+        const jobsWithLogo = jobs.map((job) => ({
+            ...job,
+            company,
+            logo,
+            link: `https://jobs.ashbyhq.com/${company}/${job.id}`,
+        }))
+
+        res.status(200).json({
+            success: true,
+            jobs: jobsWithLogo,
+        })
+    } catch (error) {
+        console.error(`API Error for ${company}:`, error)
+        res.status(500).json({
+            success: false,
+            message: `Error scraping jobs for ${company}`,
+            details: error.message,
+        })
+    }
+}
+
+const ATS_PATTERNS = [
+    {
+        name: 'Greenhouse',
+        pattern: /greenhouse\.io/i,
+        boardPattern: /greenhouse\.io\/[^/]+$/i,
+    },
+    {
+        name: 'Lever',
+        pattern: /lever\.co/i,
+        boardPattern: /lever\.co\/[^/]+$/i,
+    },
+    {
+        name: 'Workday',
```

Comment:
Workday is slightly trickier since there are so many jobs. 

```
  try {
    // Modify the URL format
    const urlParts = myWorkDayURL.split('/');
    const subdomain = urlParts[2].split('.')[0]; // e.g., 'crowdstrike' or 'disney'
    const lastPath = urlParts[urlParts.length - 1];
    const modifiedURL = `${myWorkDayURL.replace(`.com/${lastPath}`, `.com/wday/cxs/${subdomain}/${lastPath}/jobs`)}`;
    while (totalJobs === null || allJobs.length < totalJobs) {
      const response = await axios.post(
        modifiedURL,
        {
          appliedFacets: {},
          limit,
          offset,
          searchText: '',
        },
        {
          headers: {
            accept: 'application/json',
            'accept-language': 'en-US',
            'content-type': 'application/json',
            origin: new URL(myWorkDayURL).origin,
            priority: 'u=1, i',
            referer: myWorkDayURL,
            'sec-ch-ua': '"Google Chrome";v="129", "Not=A?Brand";v="8", "Chromium";v="129"',
            'sec-ch-ua-mobile': '?0',
            'sec-ch-ua-platform': '"macOS"',
            'sec-fetch-dest': 'empty',
            'sec-fetch-mode': 'cors',
            'sec-fetch-site': 'same-origin',
            'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36',
            'x-calypso-csrf-token': '1d23cde2-8ccb-4b22-baef-42bff17d22bf',
          },
        },
      );
```